src/ray/raylet/node_manager.cc

```diff
@@ -707,10 +714,61 @@ void NodeManager::ProcessClientMessage(
   client->ProcessMessages();
 }
 
+void NodeManager::ProcessActivateEventSocketRequestMessage(
+    const std::shared_ptr<LocalClientConnection> &event_client,
+    const uint8_t *message_data) {
+  auto message = flatbuffers::GetRoot<protocol::ActivateEventSocketRequest>(message_data);
+  ObjectID client_id = from_flatbuf(*message->client_id());
+  raylet_events_.ActivateEventConnection(event_client, client_id);
+  // Write the data.
+  flatbuffers::FlatBufferBuilder fbb;
+  auto activate_reply = protocol::CreateActivateEventSocketReply(fbb, true);
+  fbb.Finish(activate_reply);
+
+  event_client->WriteMessageAsync(
+      static_cast<int64_t>(protocol::MessageType::ActivateEventSocketReply),
+      fbb.GetSize(), fbb.GetBufferPointer(), [](const ray::Status &status) {
+        if (!status.ok()) {
+          // We failed to write to the client, so disconnect the client.
+          RAY_LOG(WARNING) << "Failed to send ActivateEventSocketReply to client, so "
+                              "disconnecting client";
+          // We failed to send the reply to the client, so disconnect the worker.
+          // ProcessDisconnectClientMessage(event_client);
+        }
+      });
+}
+
+void NodeManager::ProcessEventSocketMessage(
+    const std::shared_ptr<LocalClientConnection> &event_client, int64_t message_type,
+    const uint8_t *message_data) {
+  RAY_LOG(DEBUG) << "Message of type " << message_type;
+  auto message_type_value = static_cast<protocol::MessageType>(message_type);
+  switch (message_type_value) {
+  case protocol::MessageType::ActivateEventSocketRequest: {
+    ProcessActivateEventSocketRequestMessage(event_client, message_data);
+  } break;
+  case protocol::MessageType::DisconnectClient:
+  case protocol::MessageType::IntentionalDisconnectClient: {
+    // ProcessDisconnectClientMessage(event_client, /* intentional_disconnect = */ true);
```

Comment:
Can you document why we are not doing any cleanup here? Where is the cleanup happening?